pandas/io/packers.py

```diff
@@ -506,11 +526,17 @@ def decode(obj):
         dtype = dtype_for(obj[u'dtype'])
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
+
+        ctor_dtype = np_dtype
+        if is_categorical_dtype(pd_dtype):
+            # Series ctor doesn't take dtype with categorical
+            ctor_dtype = None
```

Comment:
so that's what I was first going to write. This is not a numpy-dtype and cannot be treated like one. Instead put this code in ``encode/decode`` (e.g. check ``isinstance(obj, Categorical)``). Then you won't have this issue at all.